sql/src/main/java/io/crate/operation/scalar/timestamp/CurrentTimestampFunction.java

```diff
@@ -90,17 +92,33 @@ public FunctionInfo info() {
     }
 
     @Override
-    public Symbol normalizeSymbol(Function symbol) {
-        if (symbol.arguments().isEmpty()) {
-            return Literal.newLiteral(INFO.returnType(), evaluate());
+    public Symbol normalizeSymbol(Function function, StmtCtx stmtCtx) {
+        if (stmtCtx == null) {
+            return eval(function);
         }
-        Symbol precision = symbol.arguments().get(0);
-        if (precision.symbolType().isValueSymbol()) {
-            return Literal.newLiteral(INFO.returnType(), evaluate((Input) precision));
-        } else {
-            throw new IllegalArgumentException(String.format(Locale.ENGLISH, "Invalid argument to %s", NAME));
+        // use evaluatedFunctions map to make sure multiple occurrences of current_timestamp within a query result in the same result
+        Map<Function, Symbol> evaluatedFunctions = stmtCtx.evaluatedFunctions();
+        Symbol symbol = evaluatedFunctions.get(function);
```

Comment:
i think now it could be that if you call timestamp twice in a statement with and without precission, you get different results. so i think you need to always store down to max precission and compute from there upon reuse.